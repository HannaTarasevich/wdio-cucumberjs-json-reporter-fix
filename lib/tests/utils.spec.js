import Utils from '../utils';
import { TEST_SCENARIO_STATS, TEST_SCENARIO_STATS_ERROR } from './__mocks__/mocks';

describe('utils', () => {
    describe('escapeHTML', () => {
        it('should return an empty string and escape nothing', () => {
            expect(Utils.escapeHTML('')).toEqual('');
        });

        it('should return an empty string and escape nothing', () => {
            expect(Utils.escapeHTML('<div id="feature-chart">&</div>')).toMatchSnapshot();
        });
    });

    describe('getFailedMessage', () => {
        it('should not return an error message if the status is "passed"', () => {
            expect(Utils.getFailedMessage(TEST_SCENARIO_STATS)).toEqual({});
        });

        it('should return an error message if the status is "failed"', () => {
            expect(Utils.getFailedMessage(TEST_SCENARIO_STATS_ERROR)).toMatchSnapshot();
        });
    });

    describe('containsSteps', () => {
        it('should return false if none of the values is a step name', () => {
            const steps = [
                { keyword: 'foo' },
                { keyword: 'bar' },
                { keyword: 'foobar' },
            ];
            expect(Utils.containsSteps(steps)).toEqual(false);
        });

        it('should return true if step contains the word `Given`', () => {
            const steps = [
                { keyword: 'foo' },
                { keyword: 'bar' },
                { keyword: 'Given' },
            ];
            expect(Utils.containsSteps(steps)).toEqual(true);
        });

        it('should return true if step contains the word `When`', () => {
            const steps = [
                { keyword: 'When' },
                { keyword: 'bar' },
                { keyword: 'foo' },
            ];
            expect(Utils.containsSteps(steps)).toEqual(true);
        });

        it('should return true if step contains the word `Then`', () => {
            const steps = [
                { keyword: 'bar' },
                { keyword: 'Then' },
                { keyword: 'foo' },
            ];
            expect(Utils.containsSteps(steps)).toEqual(true);
        });

        it('should return true if step contains the word `And`', () => {
            const steps = [
                { keyword: 'bar' },
                { keyword: 'And' },
            ];
            expect(Utils.containsSteps(steps)).toEqual(true);
        });
    });
});
