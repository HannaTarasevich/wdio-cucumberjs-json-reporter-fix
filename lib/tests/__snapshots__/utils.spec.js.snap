// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils getFailedMessage should return an error message if the status is "failed" 1`] = `
Object {
  "error_message": "Actual: WebdriverIO · Next-gen WebDriver test framework for Node.js
Expected: WebdriverIO

Stack:
AssertionError: expected 'WebdriverIO · Next-gen WebDriver test framework for Node.js' to equal 'WebdriverIO'
    at World.equal (/step_definitions/all.steps.js:39:35)",
}
`;
